### DISCUSSION TITLE
The extension takes Vite's WebSocket port, and currently Vite when run in middleware mode doesn't adapt, so HMR fails

### DISCUSSION THREAD

**Discussion by silvenon** [vitest-dev/vscode#313] [Feedback]
I don't consider this to be a bug of the extension, but because the extension is taking the port _all of the time_, regardless of whether the tests are running or not, I thought you should be aware of it. vitejs/vite#14328

```
WebSocket server error: Port is already in use
```

It may be a bit difficult to figure out for someone who's taking the port, so I wanted to cross-link the two projects. It's ultimately on Vite to reach a conclusion, but in order to enjoy this extension people I think thave to hard code some HMR port if they're creating Vite server in middleware mode, by using `server.hmr.port`.

Vite CLI doesn't have this problem.

**Comment by sheremet-va** [comment]
This extension always starts a Vite server in middleware mode. The problem is that Vite also _always_ starts a websocket connection even when it's not used (Vitest disables HMR).

**Reply by silvenon** [reply]
Ah, I see! That's a tough one, either way a problem upstream. Thankfully only a bummer when not using Vite CLI.
